Tooling/ScriptGenerator/ExtensionMethods.cs

```diff
@@ -0,0 +1,178 @@
+    /// </summary>
+    /// <typeparam name="T">The type of service object to get.</typeparam>
+    /// <param name="provider">The <see cref="IServiceProvider"/> to retrieve the service object from.</param>
+    /// <returns>A service object of type <typeparamref name="T"/>.</returns>
+    /// <exception cref="System.InvalidOperationException">There is no service of type <typeparamref name="T"/>.</exception>
+    [ExcludeFromCodeCoverage]
+    public static T GetRequiredServiceAndHandleError<T>(this IServiceProvider provider)
+        where T : notnull
+    {
+        try
+        {
+            return provider.GetRequiredService<T>();
+        }
+        catch (InvalidOperationException e) when (e.Message.StartsWith("Unable to resolve service for type"))
+        {
+            var currentClr = Console.ForegroundColor;
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine($"ERROR > {e.Message}");
+            Console.ForegroundColor = currentClr;
+            Environment.Exit(e.HResult);
+        }
+
+        return default!;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not the first occurence of the given <paramref name="beforeChar"/>
+    /// is located positionally before the given <paramref name="afterChar"/>.
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="beforeChar"/> and <paramref name="afterChar"/>.</param>
+    /// <param name="beforeChar">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeChar"/>.</param>
+    /// <returns><c>true</c> if the <paramref name="beforeChar"/> is located before the <paramref name="afterChar"/>.</returns>
+    public static bool IsBefore(this string value, char beforeChar, char afterChar)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            return false;
+        }
+
+        var beforeCharIndex = value.IndexOf(beforeChar);
+        var afterCharIndex = value.IndexOf(afterChar);
+
+        if (beforeCharIndex == -1)
+        {
+            return false;
+        }
+
+        if (afterCharIndex == -1)
+        {
+            return true;
+        }
+
+        return beforeCharIndex < afterCharIndex;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not the first occurence of the given <paramref name="beforeStr"/>
+    /// is located positionally before the given <paramref name="afterChar"/>.
+    /// </summary>
+    /// <param name="value">This <c>string</c> to search for the <paramref name="beforeStr"/> and <paramref name="afterChar"/>.</param>
+    /// <param name="beforeStr">The <c>char</c> that might be located before the <paramref name="afterChar"/>.</param>
+    /// <param name="afterChar">The <c>char</c> that might be located after the <paramref name="beforeStr"/>.</param>
```

Comment:
I don't like "might".   You can also use just "may be located...", or "can be located..."
```suggestion
    /// <param name="beforeChar">The <c>char</c> that may or may not be located before the <paramref name="afterChar"/>.</param>
    /// <param name="afterChar">The <c>char</c> that may or may not be located after the <paramref name="beforeChar"/>.</param>
```

Reply:
The name of the parameter will stay with the name **beforeStr**.   The acronym **Str** is short for **string** because the value is a **string** data type.  Not a **char** data type.  In this case it is important.